x/auth/ante/setup.go

```diff
@@ -37,6 +37,12 @@ func (sud SetUpContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 		return newCtx, errorsmod.Wrap(sdkerrors.ErrTxDecode, "Tx must be GasTx")
 	}
 
+	// TODO HV2: implement this check for milestones execution?
```

Comment:
Wondering if this belongs in here

Reply:
It is one of the first check we run, that's why I placed it in the outermost decorator.
Do you find any other decorator more suitable for it amongst the following?
```
		NewSetUpContextDecorator()
		NewExtensionOptionsDecorator(options.ExtensionOptionChecker),
		NewValidateBasicDecorator(),
		NewTxTimeoutHeightDecorator(),
		NewValidateMemoDecorator(options.AccountKeeper),
		NewConsumeGasForTxSizeDecorator(options.AccountKeeper),
		NewDeductFeeDecorator(options.AccountKeeper, options.BankKeeper, options.FeegrantKeeper, options.TxFeeChecker),
		NewSetPubKeyDecorator(options.AccountKeeper),
		NewValidateSigCountDecorator(options.AccountKeeper),
		NewSigGasConsumeDecorator(options.AccountKeeper, options.SigGasConsumer),
		NewSigVerificationDecorator(options.AccountKeeper, options.SignModeHandler),
		NewIncrementSequenceDecorator(options.AccountKeeper),
```

Reply:
In a fresh chain, we probably don't need this but will let @VAIBHAVJINDAL3012 confirm.

Reply:
No, we won't need it.
Just left it there to double check at the end of the migration.
We also discussed this in a call with Informal.